### PR TITLE
Add more tracing to pagination

### PR DESCRIPTION
**Pull request by jplatte**
… to help debugging timeline gap problems.

### CODE REVIEW

File: crates/matrix-sdk-ui/src/timeline/pagination.rs [hunk 1]
```diff
@@ -57,62 +71,92 @@ impl Timeline {
...
-                Err(HandleBackPaginatedEventsError::TokenMismatch) => {
-                    info!("Start of timeline was altered since pagination was started, resetting");
+            match self.paginate_backwards_once(limit, from, &mut outcome).await? {
+                PaginateBackwardsOnceResult::Success { from: None } => {
+                    trace!("Start of timeline was reached");
+                    return Ok(ControlFlow::Break(BackPaginationStatus::TimelineStartReached));
+                }
+                PaginateBackwardsOnceResult::Success { from: Some(f) } => {
+                    from = Some(f);
+                    // continue
```
Comment (poljar) [inline crates/matrix-sdk-ui/src/timeline/pagination.rs:81]: Is that supposed to be commented out? If it's supposed to be a comment and not a `continue` instruction, perhaps we can change the comment so we don't trip on this again.

**Review by poljar — APPROVED** [review]
Left two nits. Looks good otherwise.

File: crates/matrix-sdk-ui/src/timeline/pagination.rs [hunk 2]
```diff
@@ -17,11 +17,25 @@ use std::{fmt, ops::ControlFlow, pin::pin, sync::Arc, time::Duration};
...
+    /// in-flight.
+    ///
+    /// Returns `Ok(ControlFlow::Break(status))` if back-pagination succeeded,
+    /// where `status` is the resulting back-pagination status, either
+    /// [`Idle`][BackPaginationStatus::Idle] or
+    /// [`TimelineStartReached`][BackPaginationStatus::TimelineStartReached].
+    ///
+    /// Returns `Err(_)` if the a pagination request failed. This doesn't mean
+    /// that no events were added to the timeline though, it is possible that
+    /// one or more pagination requests succeeded before the failure.
```
[the remark this answers is not in the document]
Comment (jplatte) [reply crates/matrix-sdk-ui/src/timeline/pagination.rs:38]: Yeah.

File: crates/matrix-sdk-ui/src/timeline/pagination.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk-ui/src/timeline/pagination.rs:81]: Happy to apply whatever you feel makes most sense. We have another comment in back-pagination that says `// loop` in an empty match arm where the match is the only statement in a loop. I can also leave it out entirely, I don't really care much one way or the other.

Comment (poljar) [reply crates/matrix-sdk-ui/src/timeline/pagination.rs:81]: I don't know, something like:

```rust
// Fall-through here and we will continue the loop
``` 

Just so it doesn't look like an actual `continue` someone will uncomment accidentally, probably not important here even if they uncomment it, but I'd rather avoid the confusion.

**Pull request merged**